contracts/AcceleratingDistributor.sol

```diff
@@ -298,8 +267,9 @@ contract AcceleratingDistributor is ReentrancyGuard, Ownable, Multicall {
      */
     function exit(address stakedToken) external onlyInitialized(stakedToken) {
         _updateReward(stakedToken, msg.sender);
-        unstake(stakedToken, stakingTokens[stakedToken].stakingBalances[msg.sender].cumulativeBalance);
-        withdrawReward(stakedToken);
+        require(stakingTokens[stakedToken].stakingBalances[msg.sender].cumulativeBalance > 0, "Invalid amount");
```

Comment:
Consider placing this as the first line in the function, before _updateReward(). I believe it's OK, since _updateReward() doesn't touch cumulativeBalance.

Reply:
You're right thats more aligned with check-effects-interaction